spec.html

```diff
@@ -23839,17 +24442,39 @@ <h1>HostImportModuleDynamically ( _referencingScriptOrModule_, _specifier_, _pro
-            1. Let _namespace_ be GetModuleNamespace(_moduleRecord_).
-            1. If _namespace_ is an abrupt completion, perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
-            1. Else, perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
+          1. Let _stepsFulfilled_ be the steps of a CallDynamicImportFulfilled function as specified below.
+          1. Let _onFulfilled_ be CreateBuiltinFunction(_stepsFulfilled_, 0, *""*, &laquo; [[ReferencingScriptOrModule]], [[Specifier]], [[PromiseCapability]] &raquo;).
+          1. Set _onFulfilled_.[[ReferencingScriptOrModule]] to _referencingScriptOrModule_.
+          1. Set _onFulfilled_.[[Specifier]] to _specifier_.
+          1. Set _onFulfilled_.[[PromiseCapability]] to _promiseCapability_.
+          1. Let _stepsRejected_ be the steps of a CallDynamicImportRejected function as specified below.
+          1. Let _onRejected_ be CreateBuiltinFunction(_stepsRejected_, 0, *""*, &laquo; [[PromiseCapability]] &raquo;).
+          1. Set _onRejected_.[[PromiseCapability]] to _promiseCapability_.
+          1. Perform ! PerformPromiseThen(_innerPromise_, _onFulfilled_, _onRejected_).
+        </emu-alg>
+
+        <p>A CallDynamicImportFulfilled function is an anonymous built-in function with the internal slots [[ReferencingScriptOrModule]], [[Specifier]], and [[PromiseCapability]]. It takes one argument, _result_. When called, it performs the following steps:</p>
+        <emu-alg>
+          1. Assert: _result_ is *undefined*.
+          1. Let _F_ be the active function object.
+          1. Let _referencingScriptOrModule_ be _F_.[[ReferencingScriptOrModule]].
+          1. Let _specifier_ be _F_.[[Specifier]].
+          1. Let _promiseCapability_ be _F_.[[PromiseCapability]].
+          1. Let _moduleRecord_ be ! HostResolveImportedModule(_referencingScriptOrModule_, _specifier_).
+          1. Assert: Evaluate has already been invoked on _moduleRecord_ and successfully completed.
+          1. Let _namespace_ be GetModuleNamespace(_moduleRecord_).
+          1. If _namespace_ is an abrupt completion, perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
+          1. Else, perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
+        </emu-alg>
+
+        <p>A CallDynamicImportRejected function is an anonymous built-in function with one internal slot [[PromiseCapability]] and has one argument _error_. When a CallDynamicImportRejected function is called it performs the following steps:</p>
+        <emu-alg>
+          1. Let _F_ be the active function object.
+          1. Let _promiseCapability_ be _F_.[[PromiseCapability]].
+          1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _error_ &raquo;).
```

Comment:
Oh, also: since CallDynamicImport{Fulfilled,Rejected} are functions, they should return an ECMAScript language value (or a completion record), rather than just falling off the end of the algorithm.